### PR TITLE
Prepare for releasing v0.4.0

### PR DESCRIPTION
**Pull request by tenzen-y**
Mainly, I updated the release note.

Note: Blocked on #540 #543.

/assign @alculquicondor @terrytangyuan

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
(to merge at the right time)

File: RELEASE.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # MPI Operator Releases
 
+## Release v0.4.0
+* Breaking changes
+  * Removed v1 operator. If you want to use MPIJob v1, you can use the training-operator.
+* Support for the suspend semantics.
+* Fix the problem when elastic resizing workers with Horovod.
+* Support for the coscheduling plugins of the scheduler-plugins. 
+
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply RELEASE.md:9]: Uhm... in theory, Kueue support is independent of this repo.

But maybe we can say:

`Support for suspend semantics. Third party controllers can leverage the suspend field to implement queuing and preemption for an MPIJob.`

File: RELEASE.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # MPI Operator Releases
 
+## Release v0.4.0
+* Breaking changes
+  * Removed v1 operator. If you want to use MPIJob v1, you can use the training-operator.
+* Support for the suspend semantics.
+* Fix the problem when elastic resizing workers with Horovod.
```
Comment (alculquicondor) [inline RELEASE.md:7]:
```suggestion
* Bug fixes
  * Fix support for elastic Horovod.
```

**Review by terrytangyuan — APPROVED** [review]
Thanks!

/lgtm

File: RELEASE.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
 # MPI Operator Releases
 
+## Release v0.4.0
```
Comment (alculquicondor) [inline RELEASE.md:3]: Also:
- Multi arch support #489

File: RELEASE.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tenzen-y) [reply RELEASE.md:9]: I agree. Thank you for suggesting a great sentence!

[the remark this answers is not in the document]
Comment (tenzen-y) [reply RELEASE.md:11]: Done.

File: RELEASE.md (hunk 2 above)
Comment (tenzen-y) [reply RELEASE.md]: Done.

File: RELEASE.md (hunk 3 above)
Comment (tenzen-y) [reply RELEASE.md:3]: Done.

File: RELEASE.md [hunk 4]
```diff
@@ -1,5 +1,14 @@
...
+* Breaking changes
+  * Removed v1 operator. If you want to use MPIJob v1, you can use the training-operator.
+* Support for suspending semantics. Third party controllers can leverage the suspend field to implement queuing and preemption for an MPIJob.
+* Bug fixes
```
Comment (alculquicondor) [inline RELEASE.md:7]: nit: bug fixes at the end of the list

Comment (tenzen-y) [reply RELEASE.md]: Done.

**Comment by tenzen-y** [comment]
@alculquicondor Once #545 is merged into the master branch, can we merge this PR?

**Comment by alculquicondor** [comment]
Let me try to upload the images first.

**Comment by tenzen-y** [comment]
> Let me try to upload the images first.

Sure.

**Comment by tenzen-y** [comment]
Oh, conflicts are happening.

**Comment by tenzen-y** [comment]
> Oh, conflicts are happening.

Resolved.

File: Makefile [hunk 5]
```diff
@@ -17,7 +17,7 @@ REPO_PATH="github.com/kubeflow/mpi-operator"
 REL_OSARCH="linux/amd64"
 GitSHA=$(shell git rev-parse HEAD)
 Date=$(shell date "+%Y-%m-%d %H:%M:%S")
-RELEASE_VERSION?=v0.3.0
+RELEASE_VERSION?=v0.4.0
```
Comment (alculquicondor) [inline Makefile:20]: actually, we used `0.3.0` in the tag before, as opposed to `v0.3.0`

Comment (tenzen-y) [reply Makefile:20]: Upon confirmation, only `v0.2.3` has a `v` prefix.

**Comment by alculquicondor** [comment]
/lgtm

we can merge this right after the release is published in github

**Comment by alculquicondor** [comment]
or actually, before should be good

/hold cancel

**Comment by tenzen-y** [comment]
> or actually, before should be good

Previously (v0.3.0), we merged before cutting a new release.

https://github.com/kubeflow/mpi-operator/commits/v0.3.0

**Pull request merged**